core/src/main/scala/chisel3/properties/Property.scala

```diff
@@ -29,16 +38,24 @@ private[chisel3] trait PropertyType[T] {
   * be in the implicit scope and available for users.
   */
 private[chisel3] object PropertyType {
+  @implicitAmbiguous("unable to infer Property type. Please specify it explicitly in square brackets on the LHS.")
   implicit val intPropertyTypeInstance = new PropertyType[Int] {
-    override def getPropertyType: ir.PropertyType = ir.IntegerPropertyType
+    override def getPropertyType(_value: Option[Int]): ir.PropertyType = ir.IntegerPropertyType
   }
 
+  @implicitAmbiguous("unable to infer Property type. Please specify it explicitly in square brackets on the LHS.")
```

Comment:
The need for these annotations I think is indicative of my point below about not liking the fact that the only user API here is untyped. These are only necessary because the `.selectDynamic` on `Object` returns `Property[T]` but there isn't really a `T`.